internal/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,73 @@
+package dispatcher
+
+import (
+	"context"
+
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+type Dispatcher interface {
+	CheckDispatcher
+}
+
+// CheckDispatcher defines an interface that can be implemented to resolve dispatched Check
+// requests. Implementations of the interface can provide local or remote dispatching mechansims.
+type CheckDispatcher interface {
+	DispatchCheck(ctx context.Context, req *DispatchCheckRequest) (*DispatchCheckResponse, error)
+}
+
+type DispatchCheckRequest struct {
+	StoreId              string
+	AuthorizationModelId string
+	TupleKey             *openfgapb.TupleKey
+	ContextualTuples     []*openfgapb.TupleKey
+	ResolutionMetadata   *ResolutionMetadata
+}
+
+type DispatchCheckResponse struct {
+	Allowed bool
+}
+
+type ResolutionMetadata struct {
+	Depth uint32
+}
+
+func (d *DispatchCheckRequest) GetStoreId() string {
```

Comment:
These mimic protobuf generated helpers on purpose, because the idea is that this would move to a protobuf RPC definition eventually (when we implement the above interface for peer dispatching).